Feature/add django specific features

hixi opened the pull request:
Reviewed by:
- [x] @das-g

hixi committed:
add basic url patterns

hixi committed:
prepare viewset

hixi committed:
test happy path for viewset

hixi committed:
also test invalid payloads

das-g commented on tests/viewset_test.py:
```diff
@@ -0,0 +1,66 @@
+        post_viewset_result = api_client.post(utm_zone_post_url, payload, format='json')
+        expected_result = {'utm_zone_srids': [expected_srid]}
+        assert post_viewset_result.status_code == status.HTTP_200_OK
+        assert post_viewset_result.json() == expected_result
```
Consider to also assert whether the (mocked) `utm_zones_for_representing` was called with an argument reflecting the playload (incl. SRID).

das-g commented on tests/viewset_test.py:
```diff
@@ -0,0 +1,66 @@
+    dict(geom=None, srid=''),
+    dict(geom='', srid=4326),
+    dict(geom=_valid_geoJSON, srid=''),
+    dict(geom=_valid_geoJSON.copy().pop('type'), srid=4326),
```
This will pass the string `"Polygon"`. That's a valid "unhappy path" test case, but not a very interesting one. Did you intend to pass `{"coordinates": [[[100.0, 0.0], [101.0, 0.0], [101.0, 1.0], [100.0, 1.0], [100.0, 0.0]]]}` (`_valid_geoJSON` after removing `type`)?

hixi replied:
That would be better :panda_face: 
I think I thought I'd get the entire dict of type, but that's not the case, don't know what I was thinking :see_no_evil:

das-g commented on tests/viewset_test.py:
```diff
@@ -0,0 +1,66 @@
+    dict(geom='', srid=4326),
+    dict(geom=_valid_geoJSON, srid=''),
+    dict(geom=_valid_geoJSON.copy().pop('type'), srid=4326),
+    dict(geom=_valid_geoJSON.copy().pop('coordinates'), srid=4326),
```
This will pass the string `[[[100.0, 0.0], [101.0, 0.0], [101.0, 1.0], [100.0, 1.0], [100.0, 0.0]]]`. That's a valid "unhappy path" test case, but is it the one you intended, or did you mean to pass `{"type": "Polygon"}` (`_valid_geoJSON` after removing `coordinates`)?

das-g commented on utm_zone_info/viewsets.py:
```diff
@@ -0,0 +1,26 @@
+
+class UTMZoneInfoViewSet(viewsets.ViewSet):
+    """
+    A simple ViewSet for posting Points and returns valid utm_zones.
```
This is neither proper English nor factual.

das-g replied:
(fixed with 2925333 and 41e51a6)

das-g commented on utm_zone_info/viewsets.py:
```diff
@@ -0,0 +1,26 @@
+        if serializer.is_valid():
+            geometry = serializer.validated_data['geom']
+            if geometry.srid is None:
+                geometry.srid = serializer.validated_data['srid']
```
If an SRID has been explicitly passed, shouldn't it always take precedence?

hixi replied:
good catch :+1:

das-g committed:
fix grammer in UTMZoneInfoViewSet docstring

das-g committed:
more accurate UTMZoneInfoViewSet docstring

hixi committed:
make test more interesting

hixi committed:
set srid explicit

hixi committed:
Merge remote-tracking branch 'origin/feature/add_django_specific_features' into feature/add_django_specific_features

das-g committed:
use package "mock" from PyPI

so we can test on Python 2.7, too

hixi committed:
use data as test attribute, not json()

das-g committed:
actually, let's use pytest-mock

hixi committed:
test only python version we know work with certain django versions

http://stackoverflow.com/questions/34827566/attributeerror-module-html-parser-has-no-attribute-htmlparseerror

HTMLParseError is deprecated from Python 3.3 onwards and removed in Python 3.5

das-g committed:
Merge branch 'feature/add_django_specific_features' of github.com:geometalab/drf-utm-zone-info into feature/add_django_specific_features

hixi committed:
try speedup with detox on travis

hixi commented:
detox speedups on travis are minimal, but just for the sake of them I'd leave them in :-)

das-g committed:
refactoring: rename local variable for UTM zone mock SRID

It's only the expected one, because we've put it on the return value mock.

das-g committed:
refactoring: remove local variable for UTM zone mock SRID

We can get it back from the mock just fine.

das-g committed:
test that viewset calls utm_zones_for_representing with correctly constructed geometry

das-g committed:
refactoring: inline local variable for expected post call result
